Account lock after N unsuccessful login attempts

kontsevoy opened the pull request:
Very simple implementation of the account lock after N login attempts (as a non-configurable constant in defaults.go)

kontsevoy committed:
Account lock after N unsuccessful login attempts

klizhentas commented:
can you fix the test?

```
docker run --rm=true -v "$(pwd)/../":/gopath/src/github.com/gravitational/teleport -w /gopath/src/github.com/gravitational/teleport -h buildbox -u $(id -u):$(id -g) -t teleport-buildbox:latest \
	/bin/bash -c "make -C /gopath/src/github.com/gravitational/teleport TELEPORT_DEBUG=1 FLAGS='-cover -race' clean test"
make: Entering directory '/gopath/src/github.com/gravitational/teleport'
rm -rf build
go test -v github.com/gravitational/teleport/tool/tsh/... \
		   github.com/gravitational/teleport/lib/... \
		   github.com/gravitational/teleport/tool/teleport... -cover -race -tags test
# github.com/gravitational/teleport/lib/services/local
lib/services/local/services_test.go:56: not enough arguments in call to NewIdentityService
# github.com/gravitational/teleport/lib/auth
lib/auth/apiserver_test.go:99: not enough arguments in call to local.NewIdentityService
=== RUN   TestTshMain
time="0001-01-01T00:00:00Z" level=info msg="no teleport login given. defaulting to jenkins" file="client/api.go:178" 
time="0001-01-01T00:00:00Z" level=info msg="no teleport login given. defaulting to jenkins" file="client/api.go:178" 
time="0001-01-01T00:00:00Z" level=info msg="no host login given. defaulting to jenkins" file="client/api.go:185" 
time="0001-01-01T00:00:00Z" level=info msg="using FSLocalKeyStore" file="client/keystore.go:73" 
time="0001-01-01T00:00:00Z" level=info msg="SSH_AUTH_SOCK is not set. Is
```

kontsevoy commented:
done

kontsevoy committed:
Improvements to "max login attempts" feature

- increasing login attempt is now atomic within a process
- renamed some functions to be less confusing

kontsevoy committed:
Replaced "upssert" with "compare & swap"

...for login attempt counting
